pkg/config/config.go

```diff
@@ -46,7 +82,27 @@ func GetConfig(cfgFilename string) (*Config, error) {
 		return nil, fmt.Errorf("fatal error config file: %s", err)
 	}
 	var config Config
-	err = viper.Unmarshal(&config)
+	decoderConfigOption := func(dc *mapstructure.DecoderConfig) {
```

Comment:
This needs detailed comments, especially. the `if` checks.
Also, major candidate to break Retina if `DecodeHook` has bugs or missing edge cases, can you add more unit tests?